src/Core/Logger/Factory/Logger.php

```diff
@@ -19,45 +19,45 @@
  *
  */
 
-namespace Friendica\Factory;
+namespace Friendica\Core\Logger\Factory;
 
 use Friendica\Core\Config\Capability\IManageConfigValues;
-use Friendica\Core\Logger;
+use Friendica\Core\Logger\Exception\LoggerException;
+use Friendica\Core;
 use Friendica\Database\Database;
-use Friendica\Network\HTTPException\InternalServerErrorException;
 use Friendica\Util\FileSystem;
 use Friendica\Util\Introspection;
-use Friendica\Util\Logger\Monolog\DevelopHandler;
-use Friendica\Util\Logger\Monolog\IntrospectionProcessor;
-use Friendica\Util\Logger\ProfilerLogger;
-use Friendica\Util\Logger\StreamLogger;
-use Friendica\Util\Logger\SyslogLogger;
-use Friendica\Util\Logger\VoidLogger;
+use Friendica\Core\Logger\Type\Monolog\DevelopHandler;
+use Friendica\Core\Logger\Type\Monolog\IntrospectionProcessor;
+use Friendica\Core\Logger\Type\ProfilerLogger;
+use Friendica\Core\Logger\Type\StreamLogger;
+use Friendica\Core\Logger\Type\SyslogLogger;
 use Friendica\Util\Profiler;
 use Monolog;
 use Psr\Log\LoggerInterface;
 use Psr\Log\LogLevel;
+use Psr\Log\NullLogger;
 
 /**
  * A logger factory
- *
- * Currently only Monolog is supported
  */
-class LoggerFactory
+class Logger
 {
 	const DEV_CHANNEL = 'dev';
 
 	/**
 	 * A list of classes, which shouldn't get logged
 	 *
-	 * @var array
+	 * @var string[]
 	 */
 	private static $ignoreClassList = [
-		Logger::class,
+		Core\Logger::class,
 		Profiler::class,
-		'Friendica\\Util\\Logger',
+		'Friendica\\Core\\Logger\\Type',
+		'Friendica\\Core\\Logger\\Type\\Monolog',
```

Comment:
Please use the `::class` magic constant for these:
```suggestion
		\Friendica\Core\Logger\Type::class,
		\Friendica\Core\Logger\Type\Monolog::class,
```

Reply:
This works even if there is no class?

Reply:
Or, they are meant as prefixes? In this case the second line is superfluous as it is included in the more general prefix defined before.
```suggestion
		'Friendica\Core\Logger\Type',
```